### PR TITLE
Feature/ad sign in AB#93322

### PR DESCRIPTION
**Pull request by acarbutt**
### Context
Spike to intergrate with DfE Azure AD 

### Changes proposed in this pull request
Added AD authentication removed configuration login
Sign out functionaility integrated with Azure AD
Removed redundant login page

### Checklist

- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

*Cypress tests will fail, will need some further investigation if there is an option for the tests to run with the intergration

### CODE REVIEW

File: Frontend/Pages/Home/Index.cshtml.cs [hunk 1]
```diff
@@ -44,5 +49,31 @@ public async Task<IActionResult> OnGetAsync()
...
+            actionResult = null;
+            var decodedUrl = "";
+            if (!string.IsNullOrEmpty(ReturnUrl))
+            {
+                decodedUrl = WebUtility.UrlDecode(ReturnUrl);
+            }
+
+            if (Url.IsLocalUrl(decodedUrl))
+            {
+                {
```
Comment (jkempster34) [inline Frontend/Pages/Home/Index.cshtml.cs:69]: Is this double `{ { } }` accidental?

File: Frontend.Integration.Tests/IntegrationTestingWebApplicationFactory.cs [hunk 2]
```diff
@@ -44,15 +44,16 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
                 config.Sources.Clear();
                 config
                     .AddJsonFile(configPath)
+                    .AddUserSecrets<Program>()
```
Comment (jkempster34) [inline Frontend.Integration.Tests/IntegrationTestingWebApplicationFactory.cs:47]: Wondering what this does?

File: Frontend/Pages/Home/Index.cshtml.cs (hunk 1 above)
Comment (acarbutt) [reply Frontend/Pages/Home/Index.cshtml.cs]: Yep, thanks

File: Frontend.Integration.Tests/IntegrationTestingWebApplicationFactory.cs (hunk 2 above)
Comment (acarbutt) [reply Frontend.Integration.Tests/IntegrationTestingWebApplicationFactory.cs:47]: It adds the secret values for the Azure AD integration

**Review by jkempster34 — APPROVED** [review]
LGTM 💯

**Pull request merged**